### PR TITLE
ENH:  #10542 Adds __eq__() to DigMontage class and updates tests.

### LINKED ISSUE #10542: Add __eq__(), __neq__() to DigMontage
_Originally posted by @hoechenberger in https://github.com/mne-tools/mne-python/pull/10537#discussion_r851798198_

Hah. Turns out `DigMontage` inherits directly from `object` and doesn't implement its own `__eq__()`. We should probably put this on our todo.

In any case, I'll just compare `.dig` for now, which should suffice.

### PR DESCRIPTION
**Pull request by TzionaN**
#### Reference issue
Fixes #10542

#### Does:
- added '__eq__' to 'DigMontage'
- added and modifies tests

### CODE REVIEW

**Review by agramfort — COMMENTED** [review]
LGTM

@TzionaN can you add a what's new entry in latest.inc file? 🙏

**Comment by TzionaN** [comment]
Hey there!
I'm really new to contributing like this, and would love some help. 
I can't seem to understand the failed rendered and build doc tests here, since there was only a minor adjustment to the doc, and it seems to match the requested syntax. 
(Additionally, I previously recieved errors on the azure pipeline in 
mne/tests/test_epochs.py::test_base_epochs &
mne/tests/test_epochs.py::test_events_type 
which I do not understand, as nothing I did here should affect these at all, and they seem to run fine on my computer. Working on a mac - I don't know how to test Windows features.)

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -23,6 +23,8 @@ Current (1.1.dev0)
 
 Enhancements
 ~~~~~~~~~~~~
+- Add :func:`mne.channels.montage.DigMontage.__eq__` to test equality for DigMontage instances using .dig and .ch_names (:gh:`10942` by :newcontrib:`Tziona NessAiver`)
```
Comment (hoechenberger) [inline doc/changes/latest.inc:26]: Let's write it such that ordinary users will understand:
```suggestion
- :class:`~mne.channels.DigMontage` objects can now be checked for equality via ``==`` (:gh:`10942` by :newcontrib:`Tziona NessAiver`)
```

File: doc/changes/names.inc [hunk 2]
```diff
@@ -461,3 +461,5 @@
 .. _Mingjian He: https://github.com/mh105
 
 .. _Reza Nasri: https://github.com/0reza
+
+.. _Tziona NessAiver: https://github.com/TzionaN
```
Comment (hoechenberger) [inline doc/changes/names.inc:465]:
```suggestion
.. _Tziona NessAiver: https://github.com/TzionaN

```

**Comment by drammock** [comment]
> I'm really new to contributing like this, and would love some help.

Thanks for asking for help!

> I can't seem to understand the failed build doc tests here, since there was only a minor adjustment to the doc, and it seems to match the requested syntax. It seems to be related to the Azure pipeline, which I might have some difficulty troubleshooting. I'm currently working on a mac, and if I understand correctly, this is a Windows specific pipeline?

Looking at the CI runs for commit ab19cca, there were two things going on:

- `C:\Miniconda\lib\site-packages\sphinxcontrib\bibtex\transforms.py:59: PendingDeprecationWarning: nodes.Node.traverse() is obsoleted by Node.findall().` https://dev.azure.com/mne-tools/mne-python/_build/results?buildId=20802&view=logs&j=c04f1cb7-a28e-5e8a-4074-704b3a6c72b1&t=c5a3e5f7-2b1e-57e1-93fb-c99e1ea8bb1a&l=1135 This problem was unrelated to your PR, and was solved by @hoechenberger merging in changes from `main` (in other words, that particular CI problem was fixed by someone else, but your PR didn't have that fix on it until @hoechenberger merged `main` here)
- `changes/latest.inc:26: WARNING: py:func reference target not found: mne.channels.montage.DigMontage.__eq__` https://dev.azure.com/mne-tools/mne-python/_build/results?buildId=20802&view=logs&j=c04f1cb7-a28e-5e8a-4074-704b3a6c72b1&t=c5a3e5f7-2b1e-57e1-93fb-c99e1ea8bb1a&l=1968 This is a genuine problem with your PR. You cannot cross-reference any old function within MNE-Python: only functions/methods/classes that are listed in `doc/python_reference.rst` are allowable cross-reference targets.  So linking to the `__eq__` method would only work if you added it to that file (but we don't want that, we don't typically put private/builtin methods there). @hoechenberger fixed this by editing the changelog entry to not include a crossreference to `__eq__`
- The other failures (on CircleCI, e.g. [here](https://app.circleci.com/pipelines/github/mne-tools/mne-python/15353/workflows/9010abe6-edde-47ec-afe0-6d11bd89f7a7/jobs/46979?invite=true#step-139-1968)) are the same errors seen on the SphinxWindows job on Azure --- we test the same things on multiple platforms so often one mistake leads to 2 or 3 CI failures.

**Pull request merged**